addons/website/tests/test_lang_url.py

```diff
@@ -119,14 +121,15 @@ def test_01_controller_redirect(self):
         """
 
         def assertUrlRedirect(url, expected_url, msg="", code=301):
-            if expected_url.startswith('/'):
-                expected_url = "http://%s:%s%s" % (HOST, odoo.tools.config['http_port'], expected_url)
             if not msg:
                 msg = 'Url <%s> differ from <%s>.' % (url, expected_url)
 
             r = self.url_open(url, head=True)
             self.assertEqual(r.status_code, code)
-            self.assertEqual(r.headers.get('Location'), expected_url, msg)
+            parsed_location = urlparse(r.headers.get('Location', ''))
+            parsed_expected_url = urlparse(expected_url)
+            self.assertEqual(parsed_location.query, parsed_expected_url.query)
+            self.assertEqual(parsed_location.path, parsed_expected_url.path, msg)
```

Comment:
Testing the path first would produce a better error message (e.g. "/web/login" != "/my/auth/user/route"), the custom message should be used for an unequal query too
```suggestion
            self.assertEqual(parsed_location.path, parsed_expected_url.path, msg)
            self.assertEqual(parsed_location.query, parsed_expected_url.query, msg)
```